float32 msgX
float32 msgY
float32 msgZ
